src/artifactContentRenderers/textArtifactContentRenderer.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from 'vscode';
+import * as sarif from 'sarif';
+import { ArtifactContentRenderer } from './artifactContentRendering';
+
+/**
+ * Class used to render artifact content objects with text.
+ */
+export class TextArtifactContentRenderer implements ArtifactContentRenderer {
+    /**
+     * The binary string contents.
+     */
+    private readonly contents: string;
+
+    /**
+     * Creates an instance of the binary content renderer.
+     * @param content The string contents to be rendered as markdown.
+     */
+    private constructor(protected readonly artifactContent: sarif.ArtifactContent) {
+        if (!artifactContent.text) {
+            throw new Error('Expected to have text content string');
+        }
+
+        this.contents = artifactContent.text;
+    }
+
+    /**
+     * Attempts to create an instance of the binary content renderer based on a SARIF log, run index and artifact index
+     * Returns undefined if the artifact contents cannot be found, or the content is not binary content.
+     * @param log The SARIF log.
+     * @param runIndex The run index.
+     * @param artifactIndex The artifact index.
+     */
+    public static tryCreateFromLog(log: sarif.Log, artifactContents: sarif.ArtifactContent, runIndex: number, artifactIndex: number): ArtifactContentRenderer | undefined {
+        if (!artifactContents.text) {
+            return undefined;
+        }
+
+        return new TextArtifactContentRenderer(artifactContents);
+    }
+
+    /**
+     * Renders the contents as mark-down.
+     * @param artifactUri The artifact URI to use as the file-name in the markdown.
+     */
+    public render(artifactUri: vscode.Uri): string {
+        // The text property required to be UTF8 per the spec.
+        return new Buffer(this.contents).toString('utf8');
```

Comment:
You can directly return `this.contents`. Per spec, the incoming data was already utf8.

Reply:
Need to fix this in the rendering artifact renderer as well :)

---
In reply to: [411721819](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411721819) [](ancestors = 411721819)